Sprint 21

KiMate34 opened the pull request.

KiMate34 committed:
Implement library Room
Add implementation database for data-layer

KiMate34 committed:
Add repository and interactors for Library
Implement repository and interactors for Library
Add DI for repository and interactors

KiMate34 committed:
Create a layer domain, presentation and UI for database

KiMate34 committed:
Replace PlayerTrack with Track in Interactors
Replace PlayerTrack with Track in LibraryRepository
Update Room version
Remove Suspension in Flow Methods
Create PlayerTrack extension
Add trackId in PlayerTrack
Replace Duration with Long in TrackEntity
Fix request TrackDao
Delete viewModelScope in SearchScreenViewModel

KiMate34 committed:
Add listener for button ibLike
Add in execute showFavoriteEmptyList

KiMate34 committed:
Add DeleteTrackLibraryInteractor.invoke by track id

KiMate34 committed:
Now track deletes by id in TrackViewModel

KiMate34 committed:
Delete method deleteTrack by track entity

KiMate34 committed:
Change return getTrack on flow
Change method getTrackLibrary add return
Add parameter for favoriteTrackLiveData

KiMate34 committed:
Fix condition in TrackViewModel.onFavoriteClicked

KiMate34 committed:
Rename TrackDao.getTracksById and refactor TrackRepositoryImpl.searchByName

KiMate34 committed:
Fix PlayerTrack.toTrackEntity

KiMate34 committed:
Fix showing favorite track list

DmitrijEgorow commented:
- [ ] Цвет курсора не соответствует дизайну
![Screenshot_20241102_234415](https://github.com/user-attachments/assets/74c29c40-7ff2-4fed-b909-45fa5868c639)

DmitrijEgorow commented:
- [ ] Продолжительность трека отображается странно
![Screenshot_20241102_234535](https://github.com/user-attachments/assets/840ce41b-45fe-4b3c-8157-99bd9697698e)

DmitrijEgorow commented:
- [ ] Альбом может быть длинным
- [ ] Нет отступа справа у названия трека
- [ ] Кнопка лайка не соответствует дизайну
![Screenshot_20241102_234625](https://github.com/user-attachments/assets/92c015da-43c1-4a1f-9a80-5afd09fe4ef0)

DmitrijEgorow commented:
- [ ] 	❌ Если пользователь находится на экране «Аудиоплеер» и текущий трек добавлен в избранные, то пользователь видит кнопку «Нравится» в выделенном (не подсвеченном) состоянии.

https://github.com/user-attachments/assets/26ece7fc-ab0f-4f96-b574-5201086c7ac7

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/domain/entities/Track.kt:
```diff
@@ -13,4 +13,6 @@ data class Track(
     val previewUrl: String,
-)
+){
+    var isFavorite: Boolean = false
```
Лучше оставить в [data class неизменяемые свойства](https://www.reddit.com/r/Kotlin/comments/qibnqc/comment/hijhupt/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/library/ui/FavoriteTracksState.kt:
```diff
@@ -0,0 +1,11 @@
+
+import com.example.playlistmaker.player.ui.PlayerTrack
+
+interface FavoriteTracksState {
```
почему не sealed?

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/library/ui/FragmentFavoriteTracks.kt:
```diff
@@ -34,7 +44,42 @@ class FragmentFavoriteTracks : Fragment() {
+    }
+
+    private fun showFavoriteList(trackList: List<Track>) {
+        binding.favoriteListEmpty.isGone = true
+        binding.favoriteProgressBar.isGone = true
+        binding.favoriteRecyclerView.isVisible = true
```
```suggestion
        with(binding) {...}
```

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/library/ui/FragmentFavoriteTracks.kt:
[binding у фрагментов нужно занулять](https://developer.android.com/topic/libraries/view-binding#fragments)  в onDestroyView

KiMate34 committed:
Add new methods: isFavorite, onFavoriteClicked in repository

KiMate34 committed:
Add interactor OnFavoriteClickInteractor

KiMate34 committed:
Add on DI interactorModule OnFavoriteClickInteractor
Fix call method onFavoriteClicked on AudioPlayerActivity and TrackViewModel

KiMate34 committed:
Delete comment in AudioPlayerActivity
Create interactor IsFavoriteCheckInteractor, IsFavoriteCheckInteractorImpl

KiMate34 committed:
Create method FragmentFavoriteTracksViewModel.checkingFavorite
Refactor FragmentFavoriteTracks

KiMate34 committed:
Change fill color button_like_false, button_like_true ,button_in_playlist
Change style for activity audioplayer button add and like
Delete comment from AudioPlayerActivity

KiMate34 committed:
Rename context for activity_audioplayer
Add limit text for name group and track name in activity_track, activity_audioplayer
Add new drawable cursor
Add text cursor drawable for SearchEditTextStyle

DmitrijEgorow commented:
- [ ]  ❌ Если пользователь находится на экране «Аудиоплеер» и текущий трек добавлен в избранные, то пользователь видит кнопку «Нравится» в выделенном (не подсвеченном) состоянии.

https://github.com/user-attachments/assets/53c62df0-8d75-48de-b0a8-31c19d9333b1

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/library/ui/FragmentFavoriteTracksViewModel.kt:
```diff
@@ -1,7 +1,49 @@
+
+    fun getFavoriteTrackList() {
+        viewModelScope.launch{
+//            getTrackLibraryInteractor()
+//                .collect{favoriteStateMutableLiveData.postValue(checkingFavorite(it))}
```
Лучше не вливать закомментированный код

KiMate34 committed:
Delete comment in FragmentFavoriteTracksViewModel and TrackViewModel
Create method TrackViewModel.isTrackFavorite
Change life cycle scope for TrackViewModel.isTrackFavorite in AudioPlayerActivity
Add IsFavoriteCheckInteractor in DI interactorModule

KiMate34 committed:
Add style ActivityTrackTextViewPointStyle
Add string in strings.xml
Change text view tv_name_group in activity_track
Change image view on text view text_point in activity_track
Add maxLength in text view tv_name_group for activity_audioplayer

DmitrijEgorow commented:
- [ ] Нужно заново измерять view в списке, сейчас продолжительность композиции "Invention in C minor" уехала сильно вправо
- [ ] Боковые кнопки на экране плеера в светлой теме сильно светлые

https://github.com/user-attachments/assets/3c982616-354d-418b-809b-463b5bcbe247

KiMate34 committed:
Change fill alpha

KiMate34 committed:
Change wight and height tv_name_group in activity_track.xml
Add parameter ellipsize in styles.xml for activity_track.xml
